src/main/java/org/opentripplanner/model/StopPattern.java

```diff
@@ -85,19 +101,52 @@ private StopPattern (int size) {
         stops     = new Stop[size];
         pickups   = new int[size];
         dropoffs  = new int[size];
+        continuousPickup = new int[size];
```

Comment:
I went ahead and created a new object, `StopPatternFlexFields`, to hold the GTFS-Flex related fields. This will be a null field in `StopPattern` unless the feed has GTFS-Flex service.

I'm not sure why Deduplicator was used in TripTimes but not in StopPattern, but in any case, I added deduplicating in StopPattern. Only deduplicating the primitive arrays, ie not the Stops array.

Reply:
Now that I think about this more, it's probably pretty unlikely that Deduplicator would give you memory savings for StopPattern - you'd expect distinct StopPatterns to have different stop timings, otherwise OTP would just use the same StopPattern objects. That said, this is a harmless, if marginal, optimization.